File: src/main/java/net/snipsniper/scviewer/SCViewerRenderer.java
package net.snipsniper.scviewer;

import net.snipsniper.utils.Utils;
import net.snipsniper.snipscope.SnipScopeRenderer;
import net.snipsniper.snipscope.SnipScopeWindow;

import java.awt.*;

public class SCViewerRenderer extends SnipScopeRenderer {
    private final RenderingHints qualityHints = Utils.getRenderingHints();

    public SCViewerRenderer(SnipScopeWindow snipScopeWindow) {
        super(snipScopeWindow);
    }

    @Override
    public void paint(Graphics g) {
        Graphics2D g2d = (Graphics2D)g;
        g2d.setRenderingHints(qualityHints);
        super.paint(g2d);
    }
}

File: src/main/kotlin/net/snipsniper/scviewer/SCViewerRenderer.kt
package net.snipsniper.scviewer

import net.snipsniper.snipscope.SnipScopeRenderer
import net.snipsniper.snipscope.SnipScopeWindow
import net.snipsniper.utils.Utils
import java.awt.Graphics
import java.awt.Graphics2D

class SCViewerRenderer(snipSCopeWindow: SnipScopeWindow): SnipScopeRenderer(snipSCopeWindow) {
    private val qualityHints = Utils.getRenderingHints()

    override fun paint(g: Graphics) {
        g as Graphics2D
        g.setRenderingHints(qualityHints)
        super.paint(g)
    }
}